From 0d74dd7a81eb637980973a40da108a36d1153e5f Mon Sep 17 00:00:00 2001
From: Shengzhou Liu <Shengzhou.Liu@freescale.com>
Date: Thu, 2 Apr 2015 14:52:37 +0800
Subject: [PATCH 36/70] net/phy: fixup for RTL8211F link

Fix the checking link of RTL8211F.

Signed-off-by: Shengzhou Liu <Shengzhou.Liu@freescale.com>
Change-Id: I4c88f7a19f467f680957b47e0a630c950aabe46d
Reviewed-on: http://git.am.freescale.net:8181/34202
Reviewed-by: Shaohui Xie <Shaohui.Xie@freescale.com>
Tested-by: Honghua Yin <Hong-Hua.Yin@freescale.com>
Reviewed-by: Honghua Yin <Hong-Hua.Yin@freescale.com>
[czou:Original patch taken from
LS1021A-IOT-Rev2-v0.4-20150907-yocto.iso]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 drivers/net/phy/realtek.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/phy/realtek.c b/drivers/net/phy/realtek.c
index 23a5472..0860685 100644
--- a/drivers/net/phy/realtek.c
+++ b/drivers/net/phy/realtek.c
@@ -35,7 +35,7 @@
 #define RTL8211F_PHYSTAT_DUPLEX	  0x0008
 #define RTL8211F_PHYSTAT_SPDDONE  0x0800
 #define RTL8211F_PHYSTAT_LINK     0x0004
-#define PHY_AUTONEGOTIATE_TIMEOUT 50000
+#define PHY_AUTONEGOTIATE_TIMEOUT 3000
 
 MODULE_DESCRIPTION("Realtek PHY driver");
 MODULE_AUTHOR("Johnson Leung");
@@ -122,7 +122,7 @@ static int rtl8211f_config_init(struct phy_device *phydev)
 	mii_reg = phy_read(phydev, RTL8211F_PHY_STATUS);
 	phydev->link = 1;
 	while (!(mii_reg & RTL8211F_PHYSTAT_LINK)) {
-		if (i > PHY_AUTONEGOTIATE_TIMEOUT) {
+		if (i++ > PHY_AUTONEGOTIATE_TIMEOUT) {
 			pr_warn("RTL8211F LINK TIMEOUT\n");
 			phydev->link = 0;
 			break;
-- 
1.7.5.4

